src/v/kafka/server/handlers/produce.cc

```diff
@@ -336,12 +341,30 @@ static partition_produce_stages produce_topic_partition(
                       ntp,
                       source_shard);
                 }
+
+                return validator(std::move(reader))
```

Comment:
makes sense when looking at the implementation of validator::operator(), but it does read a little weird here to send everything through schema registry validator. seems ok tho

Reply:
>makes sense when looking at the implementation of validator::operator(), but it does read a little weird here to send everything through schema registry validator. seems ok tho

maybe one option would be to surface a boolean or other test for the existence of the validator here.

Reply:
Done